modules/andes-core/broker/src/main/java/org/wso2/andes/store/rdbms/RDBMSMessageStoreImpl.java

```diff
@@ -112,11 +114,34 @@ public DurableStoreConnection initializeMessageStore(AndesContextStore contextSt
         this.rdbmsStoreUtils = new RDBMSStoreUtils(connectionProperties);
         
         this.messageCache = (new MessageCacheFactory()).create();
+        initializeQueueMappingCache();
+
         log.info("Message Store initialised");
         return rdbmsConnection;
     }
 
     /**
+     * Method to initialize the queue mapping cache.
+     */
+    private void initializeQueueMappingCache() {
+        queueIDs = CacheBuilder.newBuilder()
+                .concurrencyLevel(5)
+                .maximumSize(1000)
```

Comment:
Ok. No problem. In the worst case it will read entries from DB and the memory foot print is very low.